### PR TITLE
Feature - added core schedule task models

### PR DESCRIPTION
**Pull request by gsplakmal**

### CODE REVIEW

File: src/OnlineSales/Entities/EmailSchedule.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Text.Json.Serialization;
+
+namespace OnlineSales.Entities
+{
+    public class EmailSchedule : BaseEntity
+    {
+        /// <summary>
+        /// Gets or sets the JSON based schedule assigned for the email group.
```
Comment (peterliapin) [inline src/OnlineSales/Entities/EmailSchedule.cs:14]: May be let's put some schedule sample here in the comments, so we see what types of schedules are possible?

File: src/OnlineSales/Entities/CustomerEmailSchedule.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+        [JsonIgnore]
+        [ForeignKey("CustomerId")]
+        public virtual Customer? Customer { get; set; }
+
+        /// <summary>
+        /// Gets or sets reference to EmailSchedule table.
+        /// </summary>
+        [Required]
+        public int EmaiScheduleId { get; set; }
```
Comment (peterliapin) [inline src/OnlineSales/Entities/CustomerEmailSchedule.cs:33]: Please correct typo in the name. Let's rename it simply to ScheduleId?

File: src/OnlineSales/Entities/CustomerEmailSchedule.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+        [JsonIgnore]
+        [ForeignKey("EmailScheduleId")]
```
Comment (peterliapin) [inline src/OnlineSales/Entities/CustomerEmailSchedule.cs:36]: BTW, here is is spelled right.

File: src/OnlineSales/Entities/CustomerEmailLog.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        SENT = 1,
+    }
+
+    public class CustomerEmailLog : BaseEntity
+    {
+        /// <summary>
+        /// Gets or sets reference to CustomerEmailSchedule table.
+        /// </summary>
+        [Required]
+        public int CustomerEmailScheduleId { get; set; }
```
Comment (peterliapin) [inline src/OnlineSales/Entities/CustomerEmailLog.cs:23]: let's simplify it to ScheduleId?

File: src/OnlineSales/Entities/CustomerEmailLog.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+        [JsonIgnore]
+        [ForeignKey("CustomerEmailScheduleId")]
+        public CustomerEmailSchedule? EmailSchedule { get; set; }
```
Comment (peterliapin) [inline src/OnlineSales/Entities/CustomerEmailLog.cs:27]: And here may be just Schedule?

File: src/OnlineSales/Entities/CustomerEmailLog.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+        /// <summary>
+        /// Gets or sets reference to EmailTemplate table.
+        /// </summary>
+        [Required]
+        public int EmailTemplateId { get; set; }
+
+        [JsonIgnore]
+        [ForeignKey("EmailTemplateId")]
+        public EmailTemplate? Email { get; set; }
```
Comment (peterliapin) [inline src/OnlineSales/Entities/CustomerEmailLog.cs:37]: Email -> Template

File: src/OnlineSales/Entities/CustomerEmailLog.cs [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+        [JsonIgnore]
+        [ForeignKey("CustomerEmailScheduleId")]
+        public CustomerEmailSchedule? EmailSchedule { get; set; }
+
+        /// <summary>
+        /// Gets or sets reference to EmailTemplate table.
+        /// </summary>
+        [Required]
+        public int EmailTemplateId { get; set; }
```
Comment (peterliapin) [inline src/OnlineSales/Entities/CustomerEmailLog.cs:33]: EmailTemplateId -> TemplateId

File: src/OnlineSales/Entities/CustomerEmailSchedule.cs [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        public EmailSchedule? EmailSchedule { get; set; }
```
Comment (peterliapin) [inline src/OnlineSales/Entities/CustomerEmailSchedule.cs:37]: EmailSchedule -> Schedule

File: src/OnlineSales/Entities/EmailSchedule.cs [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+        /// </summary>
+        [Required]
+        public string Schedule { get; set; } = string.Empty;
+
+        [Required]
+        public int EmailGroupId { get; set; }
```
Comment (peterliapin) [inline src/OnlineSales/Entities/EmailSchedule.cs:20]: EmailGroupId -> GroupId

File: src/OnlineSales/Entities/EmailSchedule.cs [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+        [JsonIgnore]
+        [ForeignKey("EmailGroupId")]
+        public virtual EmailGroup? EmailGroup { get; set; }
```
Comment (peterliapin) [inline src/OnlineSales/Entities/EmailSchedule.cs:24]: EmailGroup -> Group

File: src/OnlineSales/Entities/EmailTemplate.cs [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+        public string Title { get; set; } = string.Empty;
+
+        /// <summary>
+        /// Gets or sets the HTML template body of the email.
+        /// </summary>
+        [Required]
+        public string Template { get; set; } = string.Empty;
+
+        [Required]
+        public int EmailGroupId { get; set; }
```
Comment (peterliapin) [inline src/OnlineSales/Entities/EmailTemplate.cs:23]: EmailGroupId -> Group

File: src/OnlineSales/Entities/EmailTemplate.cs [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+
+        [JsonIgnore]
+        [ForeignKey("EmailGroupId")]
+        public virtual EmailGroup? EmailGroup { get; set; }
```
Comment (peterliapin) [inline src/OnlineSales/Entities/EmailTemplate.cs:27]: EmailGroup -> Group

**Review by peterliapin — CHANGES_REQUESTED** [review]

File: src/OnlineSales/Entities/EmailSchedule.cs (hunk 1 above)
Comment (gsplakmal) [reply src/OnlineSales/Entities/EmailSchedule.cs]: Updated comments

File: src/OnlineSales/Entities/CustomerEmailSchedule.cs (hunk 2 above)
Comment (gsplakmal) [reply src/OnlineSales/Entities/CustomerEmailSchedule.cs]: Renamed properly.

File: src/OnlineSales/Entities/CustomerEmailLog.cs (hunk 4 above)
Comment (gsplakmal) [reply src/OnlineSales/Entities/CustomerEmailLog.cs]: Updated naming.

File: src/OnlineSales/Entities/CustomerEmailLog.cs (hunk 5 above)
Comment (gsplakmal) [reply src/OnlineSales/Entities/CustomerEmailLog.cs]: Updated.

File: src/OnlineSales/Entities/CustomerEmailSchedule.cs (hunk 8 above)
Comment (gsplakmal) [reply src/OnlineSales/Entities/CustomerEmailSchedule.cs]: Updated.

File: src/OnlineSales/Entities/EmailSchedule.cs (hunk 9 above)
Comment (gsplakmal) [reply src/OnlineSales/Entities/EmailSchedule.cs]: Updated.

File: src/OnlineSales/Entities/EmailSchedule.cs (hunk 10 above)
Comment (gsplakmal) [reply src/OnlineSales/Entities/EmailSchedule.cs]: Updated.

File: src/OnlineSales/Entities/EmailTemplate.cs (hunk 11 above)
Comment (gsplakmal) [reply src/OnlineSales/Entities/EmailTemplate.cs]: Updated as GroupId

File: src/OnlineSales/Entities/EmailTemplate.cs (hunk 12 above)
Comment (gsplakmal) [reply src/OnlineSales/Entities/EmailTemplate.cs]: Updated

File: src/OnlineSales/Entities/CustomerEmailLog.cs (hunk 6 above)
Comment (gsplakmal) [reply src/OnlineSales/Entities/CustomerEmailLog.cs]: Updated

**Pull request merged**